<a href="https://colab.research.google.com/github/Jai-Kulkarni1905/Applied_Search_Intelligence/blob/main/work/notebooks/w01_research_question.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/Jai-Kulkarni1905/Applied_Search_Intelligence/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

My lane is Refresh and Content Opportunity Scoring. I chose this path because I am incredibly excited to build a practical decision support system that helps real SEO teams prioritize their page updates. As a student in this internship, I am absolutely eager to dive deep into feature engineering and ranking models while ensuring the final results are highly explainable.
Generating clear reason codes and actionable suggestions means I get to build something intuitive that actually solves tangible business problems, perfectly balancing my technical learning with real world impact.




*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

In [ ]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## 2. The question: decision, action, cost of a wrong call

**Decision:** Figuring out exactly which webpages should be prioritized for a content refresh or review.

**Action:** SEO managers and content strategists focus their limited time and editorial resources on pages with the highest opportunity.

**Cost of a False Positive:** Wasting valuable writing and design effort on a page that is already performing perfectly fine.

**Cost of a False Negative:** Missing out on critical organic traffic and revenue growth because a declining page went unnoticed.

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

In [ ]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [2]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.
import pandas as pd

# Load the dataset
df = pd.read_csv("content_refresh_anonymized.csv")

print(f"Total pages           : {len(df):,}")
print(f"Unique clients        : {df['client_id'].nunique()}")
print(f"Pages with 0 clicks   : {(df['clicks_90d'] == 0).sum():,} ({(df['clicks_90d'] == 0).mean()*100:.1f}%)")
print(f"Average impressions   : {df['impressions_90d'].mean():,.0f}")
print(f"Median impressions    : {df['impressions_90d'].median():,.0f}")
print(f"Average clicks        : {df['clicks_90d'].mean():.1f}")
print(f"Median clicks         : {df['clicks_90d'].median():.0f}")

Total pages           : 30,000
Unique clients        : 32
Pages with 0 clicks   : 13,204 (44.0%)
Average impressions   : 5,200
Median impressions    : 731
Average clicks        : 16.1
Median clicks         : 1


**Inference**: The dataset shows a clear imbalance in content performance, with 44% of pages receiving zero clicks and the median page receiving just 1 click despite an average of 16.1 clicks. This indicates that a small number of pages account for most of the traffic, making it impractical for SEO teams to manually review every page. This is exactly the problem my project aims to solve by ranking pages based on their refresh opportunity, allowing content teams to focus their efforts on pages that are most likely to benefit from an update rather than relying on manual inspection or intuition.

## 4. Careful words: what I can and can't claim

This project uses historical search data to give SEO teams smart, evidence-based recommendations on which pages are most ripe for a refresh.

The project does not claim to crack Google’s algorithm or guarantee future rankings, as outside factors like seasonality and competitor moves are always at play.

This project operates strictly as a directional tool to help prioritize efforts and support human intuition, rather than functioning as an automated system.

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

In [ ]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.

## Self-check

Before you submit, confirm each line honestly:

-  Every section above is filled — markdown thinking AND the code that backs it
-  The notebook runs top to bottom with no errors (Runtime → Run all)
-  No client names, URLs, or private queries anywhere
-  My claims use careful words: observed, measured, directional, decision-support
-  Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.